e2e/nomostest/clean.go

```diff
@@ -126,6 +133,88 @@ func Clean(nt *NT) error {
 	return deleteImplicitNamespacesAndWait(nt)
 }
 
+func uninstallAllPackages(nt *NT) error {
+	// Check if reconciler-manager is reconciled and healthy
+	nt.T.Log("[CLEANUP] checking reconciler-manager status...")
+	rmObj := &unstructured.Unstructured{}
+	rmObj.SetGroupVersionKind(kinds.Deployment())
+	rmObj.SetNamespace(configmanagement.ControllerNamespace)
+	rmObj.SetName(reconcilermanager.ManagerName)
+	if err := nt.KubeClient.Get(rmObj.GetName(), rmObj.GetNamespace(), rmObj); err != nil {
+		return fmt.Errorf("checking reconciler-manager status: %v", err)
+	}
+	if err := testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus)(rmObj); err != nil {
+		return fmt.Errorf("reconciler-manager not healthy: %v", err)
+	}
+	// If reconciler-manager is healthy, uninstall all the unmanaged RootSyncs & RepoSyncs.
+	// If deletion-propagation is enabled, this will also recursively delete all
+	// managed objects, including other managed packages.
+	if err := uninstallUnmanagedPackagesOfType(nt, kinds.RepoSyncV1Beta1()); err != nil {
+		return fmt.Errorf("uninstalling unmanged RepoSyncs: %v", err)
+	}
+	if err := uninstallUnmanagedPackagesOfType(nt, kinds.RootSyncV1Beta1()); err != nil {
+		return fmt.Errorf("uninstalling unmanged RootSyncs: %v", err)
+	}
+	return nil
+}
+
+func uninstallUnmanagedPackagesOfType(nt *NT, gvk schema.GroupVersionKind) error {
+	// Find all packages
+	nt.T.Logf("[CLEANUP] listing %s objects...", gvk.Kind)
+	rsObjList := &unstructured.UnstructuredList{}
+	rsObjList.SetGroupVersionKind(kinds.ListGVKForItemGVK(gvk))
+	if err := nt.KubeClient.List(rsObjList, client.InNamespace(configmanagement.ControllerNamespace)); err != nil {
+		return fmt.Errorf("listing %s: %v", gvk.Kind, err)
+	}
+	// Find the unmanaged packages
+	var rsObjs []*unstructured.Unstructured
+	for _, item := range rsObjList.Items {
+		if core.GetAnnotation(&item, metadata.ResourceManagementKey) == metadata.ResourceManagementEnabled {
```

Comment:
I think the intent is `core.GetAnnotation(&item, metadata.ResourceManagementKey) != metadata.ResourceManagementEnabled` to get *unmanaged* objects

Reply:
Fixed.